.github/workflows/update_sdk_methods.py

```diff
@@ -662,11 +691,44 @@ def parse(type, names):
 
                                 ## Get full method usage string, by omitting all comment spans:
                                 method_usage_raw = regex.sub(r'<span class="comment">.*</span>', '', this_method_raw2)
-                                this_method_dict["usage"] = regex.sub(r'</span>', '', method_usage_raw).lstrip().rstrip()
+                                this_method_dict["usage"] = regex.sub(r'</span>', '', method_usage_raw).replace("\t", "  ").lstrip().rstrip()
 
                                 ## Not possible to link to the specific functions, so we link to the parent resource instead:
                                 this_method_dict["method_link"] = url + '#' + interface_name
 
+                                ## Check for code sample for this method.
+                                ## If we detected that a local instance of the Go SDK docs is available, use that.
+                                ## Otherwise, use the existing scraped 'soup' object from the live Go SDK docs instead.
+                                if is_go_sdk_staging_available:
+
+                                    staging_url = regex.sub('https://pkg.go.dev', 'http://localhost:8080', url)
+                                    staging_soup = make_soup(staging_url)
+
+                                    ## Get a raw dump of all go methods by interface for each resource:
+                                    go_code_samples_raw = staging_soup.find_all(
+                                        lambda code_sample_tag: code_sample_tag.name == 'p'
+                                        and method_name + " example:" in code_sample_tag.text)
+                                else:
+
+                                    ## Get a raw dump of all go methods by interface for each resource:
+                                    go_code_samples_raw = soup.find_all(
+                                        lambda code_sample_tag: code_sample_tag.name == 'p'
+                                        and method_name + " example:" in code_sample_tag.text)
+
+                                ## Determine if a code sample is provided for this method:
+                                if len(go_code_samples_raw) == 1:
+
+                                    ## Fetch code sample raw text, preserving newlines but stripping all formatting.
+                                    ## This string should be suitable for feeding into any python formatter to get proper form:
+                                    this_method_dict["code_sample"] = go_code_samples_raw[0].find_next('pre').text.replace("\t", "  ")
+
+                                elif len(go_code_samples_raw) > 1:
+                                
+                                    ## In case we want to support multiple code samples per method down the line,
```

Comment:
Haha I realize, though, that in leaving room for future multiple samples here, I am in fact no longer considering _any_ code samples if more than one are ever detected. I should duplicate line 723 here as well, so that, even if two are detected, we'll still get at least the first one regardless.